xir/program.py

```diff
@@ -167,9 +195,9 @@ def pref(self) -> Union[Decimal, float, int, str]:
         return self._pref
 
     @property
-    def terms(self) -> List:
+    def factors(self) -> List:
```

Comment:
Assuming that `self._factors` is always a list,
```suggestion
    def factors(self) -> List[ObservableStmt]:
```
Otherwise, if it is a sequence, we can write
```suggestion
    def factors(self) -> Sequence[ObservableStmt]:
```

Reply:
Yeah, we should avoid over-constraining types where possible. This results in a more flexible interface so that
```python
def avg(s: Sequence[int]) -> int:
    return sum(s) / len(s)
```
can be called using
```python
avg([0, 1, 2, 3, 4])
```
or
```python
avg((0, 1, 2, 3, 4))
```
or even
```python
avg(range(5))
```
[This page](https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes) is a pretty handy reference.